addons/udes_stock/models/mixin_stock_model.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+
+
+class MixinStockModel(models.AbstractModel):
+    _name = "mixin.stock.model"
+    _description = "Inherited Model"
+
+    # Whether the model is allowed to create a record when using get_or_create
+    MSM_CREATE = False
+    # Default fields to search with a str on get_or_create
+    MSM_STR_DOMAIN = ("name",)
+
+    def _get_msm_domain(self, identifier):
```

Comment:
Nit: Could we add a doc-string to this method?

Reply:
I don't know the context behind this function (other than the variables it used causing some issues in tests). To help write the docstring context on why it exists would be useful

Reply:
am thinking something like
```python
     def _get_msm_domain(self, identifier):
        """Return a domain based on MSM_STR_DOMAIN attribute
        (which can be changed on the model this MixinStockModel is mixed in to"""
```

Reply:
That looks good to me :) Only super nit is that Odoo should be capitalised, sorry 😅